### ISSUE TITLE
Add Mixer Check support for health-checks

### ISSUE THREAD

**Issue by douglas-reid** [istio/istio#2808]
cc: @ZackButcher 

Needs more investigation, but currently reported as:

Envoy calls to mixer.Check() are failing for health-checks during pod start-up. Suspected cause: no `destination.service` attribute available.

Some potential mitigations:
1) add pre-process adapter for detecting scenario and using a special `destination.service` value (maybe `healthcheck`). this assumes we can execute the adapter before failure.
2) short-circuit Mixer call before adapter dispatch.

Updates here as more info is discovered.

**Comment by douglas-reid** [comment]
initially assigning to @ZackButcher for investigation. when root cause is positively identified, this will be reassigned for fix as appropriate.

**Comment by ayj** [comment]
Is there a design/plan around bypassing aspects of a Check() call for health checks? For example, health checks made by the kubelet probably ought not contribute to a service's quota limit. Can we differentiate based on source?

**Comment by ZackButcher** [comment]
I'm still root causing before I get a design together. In fact, it appears that the implementation notes in the pilot code base don't line up with the implementation itself, and we do in fact call Mixer `Check` for liveliness/health checks when those endpoints are on the port as the server itself. Further, those `Check`s appear to pass just fine.

I'll get together a small case to prove things out.

**Comment by mandarjog** [comment]
We should surely exclude readiness probes from mixer checks. We should be able to read the pod spec for readiness probe configuration. As for tcp liveness checks, we can check ip address of the kublet, but that seems problematic.

**Comment by ZackButcher** [comment]
The problem is when health/liveliness checks occur on the same port as the main app. Deciding when to bypass in that case is hard (or impossible).

**Comment by mandarjog** [comment]
`if path == "/health" then bypass()`   unless this is a tcp connection :-)

Does kubelet have the correct certs to connect to the application ?

**Comment by ZackButcher** [comment]
No, the assumption today is no certs for the health checking service (kube API server in this case). The case where the mesh has mTLS enabled but health checks without a cert is blocked on us landing TLS sniffing.

Anything like white-listing specific paths, etc. needs to be configurable per service, since k8s has no specific requirements on that front (you just tell it "make a GET request to this URI"). Ideally we could [pull it from the deployment](https://github.com/ZackButcher/test-server/blob/40bab4136fcd142bc257ab5360deb0326efacd31/kubernetes/deployment.yaml#L28-L39) itself (but that's k8s specific).
